src/sobol_sensitivity.jl

```diff
@@ -342,10 +342,18 @@ function gsa_sobol_all_y_analysis(method, all_y::AbstractArray{T}, d, n, Ei_esti
         nboot > 1 ? reshape(ST_CI, size_...) : nothing)
 end
 
-function gsa(f, method::Sobol, p_range::AbstractVector; samples, kwargs...)
+function gsa(f, method::Sobol, p_range::AbstractVector; samples,
+        rng::AbstractRNG = Random.default_rng(), kwargs...)
+    log2num = ceil(Int, log2(samples))
+    samples2n = 2^log2num
+    if samples2n != samples
+        samples = samples2n
+        @warn "Passed samples is not a power of 2, number of sample points changed to $samples"
+    end
     AB = QuasiMonteCarlo.generate_design_matrices(samples, [i[1] for i in p_range],
         [i[2] for i in p_range],
-        QuasiMonteCarlo.SobolSample(),
+        QuasiMonteCarlo.SobolSample(;
+            R = QuasiMonteCarlo.OwenScramble(; base = 2, pad = log2num, rng)),
```

Comment:
Any `pad` >= `log2num` should work AFAICT - in the QuasiMonteCarlo docs commented out versions use e.g. also `2 * log2num` but I don't know if that is a (significant) improvement.

Reply:
I think it would also be good to change the examples and tests accordingly.